cli/src/cli/addorder/addorder.rs

```diff
@@ -0,0 +1,100 @@
+use ethers::types::{H160, U256,Bytes};
+use ethers_signers::Ledger;
+use spinners::{Spinner, Spinners};
+use std::str::FromStr;
+use std::{convert::TryFrom, sync::Arc};
+use ethers:: prelude::SignerMiddleware;
+use ethers::providers::{Provider, Http} ;
+
+use crate::cli::registry::{IOrderBookV2, IParserV1, Io, EvaluableConfig, OrderConfig};
+
+#[allow(unused_variables)]
+pub async fn add_ob_order(
+    orderbook_address : H160,
+    parser_address : H160,
+    tokens : Vec<String>,
+    decimals : Vec<u8>,
+    order_string: String ,
+    order_meta : String ,
+    rpc_url : String,
+    wallet : Ledger
+) -> anyhow::Result<()> {
+
+    let provider = Provider::<Http>::try_from(rpc_url)
+    .expect("\n❌Could not instantiate HTTP Provider");   
+
+    let client = SignerMiddleware::new_with_provider_chain(provider.clone(), wallet).await?; 
+
+    let orderbook = IOrderBookV2::new(orderbook_address, Arc::new(client)); 
+
+    let parser_contract = IParserV1::new(parser_address.clone(),Arc::new(provider.clone())) ;   
+
+    let (sources,constants) = parser_contract.parse(
+        Bytes::from(
+            order_string.as_bytes().to_vec()
+        )
+    ).call().await.unwrap() ; 
+
+    let tokens = tokens ;
+    let decimals = decimals ;
+
+    let vault_id = U256::from(H160::random().as_bytes()) ;   
```

Comment:
this needs to be an input otherwise we can't have 2 orders pointing to the same vault, which is needed to buy/sell in a loop